tests/defer/src/commonTest/kotlin/test/DeferTest.kt

```diff
@@ -0,0 +1,271 @@
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", null),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", null),
+                )
+            )
+        ).build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", null),
+                )
+            )
+        ).build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", null),
+                )
+            )
+        )
+            .errors(
+                listOf(
+                    Error(
+                        message = "Cannot resolve isColor",
+                        locations = listOf(Error.Location(1, 119)),
+                        path = listOf("computers", 0, "screen", "isColor"),
+                        extensions = null, nonStandardFields = null
+                    )
+                )
+            )
+            .build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", ComputerFields("486", 1996,
+                        ComputerFields.Screen("Screen", "800x600", null))),
+                )
+            )
+        ).build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", ComputerFields("486", 1996,
+                        ComputerFields.Screen("Screen", "800x600",
+                            ScreenFields(true)))),
+                )
+            )
+        ).build(),
+    )
+
+    mockServer.enqueueMultipart(jsonList)
+    val actualResponseList = apolloClient.query(query).toFlow().toList()
+    assertResponseListEquals(expectedDataList, actualResponseList)
```

Comment:
Nice tests! 👍 

It'd be interesting to test against a real server too. I think helix has defer support. We could make a JS-only test that runs helix? Like we have JVM-only tests running graphql-kotlin ?

Reply:
Totally! I've been using Helix locally to test stuff. Not sure how hard it would be to integrate it inside Kotlin tests but that definitely would be useful! I'll create a specific ticket.